### PR TITLE
Windows build: needed packages update

### PR DESCRIPTION
**Pull request by michaelschufi**
Since I reinstalled msys2 recently, I went through the Windows build process again. I found that you need the following msys2 packages:
- mingw-w64-x86_64-gcc
- mingw-w64-x86_64-cmake
- mingw-w64-x86_64-make
- patch (for the lua setup)

### CODE REVIEW

**Comment by Technius** [comment]
Our windows readme is out of date and not consistent with our automated builds:
https://github.com/xournalpp/xournalpp/blob/e2f90ce10b7ce631a418c112551f637ef80d795a/azure-pipelines/release.yml

Can you also update the relevant build files? You should only need to edit `azure-pipelines/release.yml` and `azure-pipelines/CI.yml`

File: .gitignore [hunk 1]
```diff
@@ -49,3 +49,6 @@ repo/
 flatpak-build/
 .flatpak-builder/
 *.flatpak
+
+# LUA build
+lua-5.3.5/
```
Comment (LittleHuba) [inline .gitignore]: Please use `lua-*/` instead, as we risk missing this at a version bump...

Comment (michaelschufi) [reply .gitignore]: did that :)

**Comment by michaelschufi** [comment]
> 
> 
> Our windows readme is out of date and not consistent with our automated builds: ...
> 
> Can you also update the relevant build files? You should only need to edit `azure-pipelines/release.yml` and `azure-pipelines/CI.yml`

Do you want me to update the windows readme and the azure files or just the azure files?

**Comment by LittleHuba** [comment]
Both please!

**Comment by LittleHuba** [comment]
I'll merge the branch on an internal branch and optimize a few things. Thanks for your work!

**Pull request merged**